src/framework/audio/internal/platform/win/wincoreaudiodriver.cpp

```diff
@@ -45,93 +59,99 @@ std::string CoreAudioDriver::name() const
-    LOGI() << pFormat.nSamplesPerSec;
-    pFormat.wBitsPerSample = 32;
-    pFormat.nAvgBytesPerSec = pFormat.nSamplesPerSec * pFormat.nChannels * sizeof(float);
-    pFormat.nBlockAlign = (pFormat.nChannels * pFormat.wBitsPerSample) / 8;
-    pFormat.cbSize = 0;
+    s_data->pFormat = *deviceFormat;
+    s_data->pFormat.wFormatTag = WAVE_FORMAT_IEEE_FLOAT;
+    s_data->pFormat.nChannels = spec.channels;
+    LOGI() << s_data->pFormat.nSamplesPerSec;
+    s_data->pFormat.wBitsPerSample = 32;
+    s_data->pFormat.nAvgBytesPerSec = s_data->pFormat.nSamplesPerSec * s_data->pFormat.nChannels * sizeof(float);
+    s_data->pFormat.nBlockAlign = (s_data->pFormat.nChannels * s_data->pFormat.wBitsPerSample) / 8;
+    s_data->pFormat.cbSize = 0;
+    s_data->callback = spec.callback;
     if (activeSpec) {
         *activeSpec = spec;
         activeSpec->format = Format::AudioF32;
-        activeSpec->sampleRate = pFormat.nSamplesPerSec;
+        activeSpec->sampleRate = s_data->pFormat.nSamplesPerSec;
     }
-    hr = m_audioClient->Initialize(AUDCLNT_SHAREMODE_SHARED, AUDCLNT_STREAMFLAGS_EVENTCALLBACK,
-                                   defaultTime, defaultTime, &pFormat, NULL);
+    hr = s_data->audioClient->Initialize(AUDCLNT_SHAREMODE_SHARED, AUDCLNT_STREAMFLAGS_EVENTCALLBACK,
+                                   defaultTime, defaultTime, &s_data->pFormat, NULL);
     CHECK_HRESULT(hr);
 
     UINT32 bufferFrameCount;
-    hr = m_audioClient->GetBufferSize(&bufferFrameCount);
+    hr = s_data->audioClient->GetBufferSize(&bufferFrameCount);
     CHECK_HRESULT(hr);
 
-    hr = m_audioClient->GetService(IID_IAudioRenderClient, (void**)&m_renderClient);
+    hr = s_data->audioClient->GetService(MU_IID_IAudioRenderClient, (void**)&s_data->renderClient);
     CHECK_HRESULT(hr);
 
-    auto hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+    s_data->hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
     CHECK_HRESULT(hr);
-    hr = m_audioClient->SetEventHandle(hEvent);
+    hr = s_data->audioClient->SetEventHandle(s_data->hEvent);
 
     m_active = true;
-    m_thread = std::thread([=]() {
+    m_thread = std::thread([this]() {
         BYTE* pData;
         HRESULT hr = S_OK;
         do {
             UINT32 bufferFrameCount, bufferPading;
-            hr = m_audioClient->GetBufferSize(&bufferFrameCount);
+            hr = s_data->audioClient->GetBufferSize(&bufferFrameCount);
             logError(hr);
 
-            hr = m_audioClient->GetCurrentPadding(&bufferPading);
+            hr = s_data->audioClient->GetCurrentPadding(&bufferPading);
             logError(hr);
 
             auto bufferSize = bufferFrameCount - bufferPading;
-            hr = m_renderClient->GetBuffer(bufferSize, &pData);
+            hr = s_data->renderClient->GetBuffer(bufferSize, &pData);
             logError(hr);
             if (!pData || hr != S_OK) {
                 continue;
             }
-            activeSpec->callback(nullptr, reinterpret_cast<uint8_t*>(pData),
```

Comment:
This is the first error - accessing this pointer in the thread, but this structure is an output argument and is a local variable, and after opening the driver, it will be destroyed, so an invalid pointer is being accessed in the thread.